src/moments.jl

```diff
@@ -161,87 +161,57 @@ end
 
 ##### General central moment
 function _moment2(v::AbstractArray{<:Real}, m::Real; corrected=false)
-    n = length(v)
-    s = 0.0
-    for i = 1:n
-        @inbounds z = v[i] - m
-        s += z * z
-    end
-    varcorrection(n, corrected) * s
+    s = sum(x->abs2(x-m), v, init=zero(m))
+    return varcorrection(length(v), corrected) * s
 end
 
 function _moment2(v::AbstractArray{<:Real}, wv::AbstractWeights, m::Real; corrected=false)
-    n = length(v)
-    s = 0.0
-    for i = 1:n
-        @inbounds z = v[i] - m
-        @inbounds s += (z * z) * wv[i]
-    end
-
-    varcorrection(wv, corrected) * s
+    s = sum(i -> (@inbounds abs2(v[i] - m) * wv[i]), eachindex(v), init=zero(m))
```

Comment:
Similarly, I think we need a different `init` here. Additionally, `@inbounds` is unsafe here as `i` might not be an actual index of `wv`.

Reply:
```suggestion
    init = (zero(eltype(v)) - zero(m))^2 * zero(eltype(wv))
    s = sum(i -> (v[i] - m)^2 * wv[i], eachindex(v, wv); init=init)
```

Reply:
@devmotion if `v` and `wv` do not have the same indices `eachindex(v, wv)` will throw and therefore it seems that `@inbounds` here should be safe. How about this:
```suggestion
    s = if isempty(v)
         (zero(eltype(v)) - zero(m))^2*zero(eltype(wv))
    elseif eachindex(v) == eachindex(wv)
        sum(i -> (@inbounds (v[i] - m)^2 * wv[i]), eachindex(v, wv))
    else
        sum(i -> ((v_ - m)^2 * wv_) for (v_, wv_) in zip(v, wv))
    end
```

Reply:
@devmotion bump

Reply:
@andreasnoack I don't want to be negative (I'm really writing this with a smile :D ), but given the pace we had with github.com/JuliaStats/HypothesisTests.jl/pull/124 I expect this to be merged ~ 2030 :D

But by any means I'm not in a hurry.